File: frontend/app/types/Problem.ts
export type Problem = {
    id: number;
    title: string;
    description: string;
    difficulty: 'Easy' | 'Medium' | 'Hard'
    topic: string;
    companies: string;
    boilerplate_PY: string;
    boilerplate_JS: string;
    boilerplate_CPP: string;
}

File: backend/internal/models/problem.go
package models

type Problem struct {
	Id              int    `json:"id"`
	Title           string `json:"title"`
	Difficulty      string `json:"difficulty"`
	CreatedAt       string `json:"created_at"`
	Description     string `json:"description"`
	Topic           string `json:"topic"`
	Companies       string `json:"companies"`
	Boilerplate_PY  string `json:"boilerplate_PY"`
	Boilerplate_JS  string `json:"boilerplate_JS"`
	Boilerplate_CPP string `json:"boilerplate_CPP"`
}
